File: src/main/java/com/github/kory33/signvote/exception/InvalidScoreVotedException.java
package com.github.kory33.signvote.exception;

import com.github.kory33.signvote.vote.VotePoint;
import com.github.kory33.signvote.vote.VoteScore;
import lombok.EqualsAndHashCode;
import lombok.Value;
import org.bukkit.entity.Player;

/**
 * Represents an exception thrown when a player attempts to vote with an invalid score
 */
@EqualsAndHashCode(callSuper = false)
@Value
public class InvalidScoreVotedException extends Exception {
    VotePoint targetVotePoint;
    Player voter;
    VoteScore voteScore;
}

File: src/main/java/com/github/kory33/signvote/exception/InvalidScoreVotedException.kt
package com.github.kory33.signvote.exception

import com.github.kory33.signvote.vote.VotePoint
import com.github.kory33.signvote.vote.VoteScore
import org.bukkit.entity.Player

/**
 * Represents an exception thrown when a player attempts to vote with an invalid score
 */
data class InvalidScoreVotedException
constructor(val targetVotePoint: VotePoint, val voter: Player, val voteScore: VoteScore) : Exception()
